pkg/hope/node_management.go

```diff
@@ -303,3 +319,48 @@ func forceUserToEnterHostnameToContinue(node *Node) error {
 
 	return nil
 }
+
+func applyNodeRegistrationTaints(node *Node) error {
+	if len(node.Taints) == 0 {
+		return nil
+	}
+
+	// Can't just scp, because of permissions.
+	yamlContent, err := ssh.GetSSH(node.ConnectionString(), "sudo", "cat", kubeletConfigPath)
+	if err != nil {
+		return err
+	}
+
+	yamlReader := strings.NewReader(yamlContent)
+	var kubeletConfig kubeletconfigv1betal.KubeletConfiguration
+	kubernetesyaml.NewYAMLOrJSONDecoder(yamlReader, 1000).Decode(&kubeletConfig)
```

Comment:
Or rather, should try to see if the other yaml package can handle this. We know that this is going to be a yaml file already, don't need to be using "maybe it's JSON, maybe it's YAML" mentality.